### PR TITLE
feat: Add BeanReferenceField

### PR DESCRIPTION
**Pull request by igarashitm**
Fixes: #470

https://github.com/KaotoIO/kaoto-next/assets/265462/af604413-305e-42ab-ba4d-b575f0e7462f

### CODE REVIEW

File: packages/ui/src/components/Form/bean/BeanReferenceField.test.tsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+      );
+    });
+    const toggleButton = screen.getByRole('button', { name: 'Menu toggle' });
+    await act(async () => {
+      fireEvent.click(toggleButton);
+    });
+    const createNewOption = screen.getAllByRole('option').filter((option) => option.textContent === 'Create new bean');
+    expect(createNewOption).toHaveLength(1);
+    await act(async () => {
+      //  TypeError: Cannot read properties of undefined (reading 'Provider')
```
Comment (igarashitm) [inline packages/ui/src/components/Form/bean/BeanReferenceField.test.tsx:137]: I tried to test the "create new bean" modal is rendered, but the weird error was thrown from inside the CustomAutoForm. This test is skipped for now.

**Comment by lordrip** [comment]
I like this placeholder :tada: 
image

File: packages/ui/src/components/Form/bean/BeanReferenceField.test.tsx (hunk 1 above)
Comment (lordrip) [reply packages/ui/src/components/Form/bean/BeanReferenceField.test.tsx:156]: @igarashitm I created this https://github.com/igarashitm/kaoto-next/pull/4 to enable this test.

The **TL;DR** is that `uniforms` fields require a `componentDetector` provider, and to be a descendant of `AutoForm` which holds the schema bridge.

The stack trace you saw is complaining about a nested field without a detector.

**Review by lordrip — CHANGES_REQUESTED** [review]
@igarashitm it looks great, definitively this is a game changer.

I'm marking it as `Request changes` because I think there's a little detail that's unexpected IMHO, adding new beans it's only possible when there's an existing `Beans` array, otherwise, only the name gets reflected into the property.

In case this is not expected, here are the steps to reproduce:
1. Remove the `Beans` array from the code
2. Go to the form and create a new `Bean`
3. After saving the new bean, you'll see the property updated but no bean created

If you try this with an existing `Beans` array, then it works fine.

File: packages/ui/src/components/Form/bean/BeanReferenceField.test.tsx (hunk 1 above)
Comment (igarashitm) [reply packages/ui/src/components/Form/bean/BeanReferenceField.test.tsx]: Thanks! I never thought the outer form is the cause as the error message looked like coming from the inner form, the new bean form in the modal :sweat_smile:

**Comment by igarashitm** [comment]
> adding new beans it's only possible when there's an existing `Beans` array, otherwise, only the name gets reflected into the property.

OMG thanks for catching this, I totally forgot to test this case :scream:

**Comment by igarashitm** [comment]
Done, and added a test for that - https://github.com/KaotoIO/kaoto-next/pull/552/files#diff-211e7d13ad90f7d65608d307de7d9d1673727b8cbdcc0ace8e2252020830ca4eR266

**Review by lordrip — APPROVED** [review]

**Pull request merged**